spec-tests/src/test/java/org/candlepin/spec/jobs/InactiveConsumerCleanerJobTest.java

```diff
@@ -0,0 +1,157 @@
+
+    private static final String JOB_KEY = "InactiveConsumerCleanerJob";
+    private static final int DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS = 397;
+
+    private ConsumerApi consumerApi;
+    private OwnerApi ownerApi;
+    private OwnerProductApi ownerProductApi;
+    private JobsApi jobsApi;
+    private ConsumerTypeApi consumerTypeApi;
+
+    private OwnerDTO owner;
+    private ConsumerTypeDTO consumerType;
+
+    @BeforeEach
+    private void beforeEach() throws ApiException {
+        ApiClient client = ApiClients.admin();
+        consumerApi = client.consumers();
+        ownerApi = client.owners();
+        ownerProductApi = client.ownerProducts();
+        jobsApi = client.jobs();
+        consumerTypeApi = client.consumerTypes();
+
+        if (owner == null) {
+            owner = ownerApi.createOwner(Owners.random());
+        }
+
+        if (consumerType == null) {
+            ConsumerTypeDTO newType = new ConsumerTypeDTO();
+            newType.setManifest(false);
+            newType.setLabel(StringUtil.random("Test Type"));
+            consumerType = consumerTypeApi.createConsumerType(newType);
+        }
+    }
+
+    @Test
+    public void shouldDeleteInactiveConsumers() throws Exception {
+        Instant inactiveTime = Instant.now()
+            .minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS + 10, ChronoUnit.DAYS);
+        Instant activeTime = Instant.now()
+            .minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS - 10, ChronoUnit.DAYS);
+
+        ConsumerDTO inactiveConsumer = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(inactiveTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO consumerWithEntitlement = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(inactiveTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO activeConsumer = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(activeTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO activeConsumer2 = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(activeTime.atOffset(ZoneOffset.UTC))
+            .build();
+
+        inactiveConsumer = consumerApi.createConsumer(inactiveConsumer, null, owner.getKey(), null, false);
+        consumerWithEntitlement = consumerApi
+            .createConsumer(consumerWithEntitlement, null, owner.getKey(), null, false);
+        activeConsumer = consumerApi.createConsumer(activeConsumer, null, owner.getKey(), null, false);
+        activeConsumer2 = consumerApi.createConsumer(activeConsumer2, null, owner.getKey(), null, false);
+
+        createProductAndPoolForConsumer(consumerWithEntitlement);
+
+        AsyncJobStatusDTO status = jobsApi.scheduleJob(JOB_KEY);
+        boolean isComplete = JobUtil.waitForJobToComplete(status.getId(), jobsApi);
+        assertTrue(isComplete);
```

Comment:
Can we also assert that the job's status is `FINISHED` (because it could complete with a different state, like FAILED)?